aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -1,9 +1,128 @@
+    - https://docs.aws.amazon.com/firehose/latest/dev/data-transformation.html
+    """
+
+    # Record ID; uniquely identifies this record within the current batch"""
+    record_id: str
+    # Processing result, supported value: Ok, Dropped, ProcessingFailed"""
+    result: Literal["Ok", "Dropped", "ProcessingFailed"]
+    # data blob, base64-encoded, optional at init. Allows pass in base64-encoded data directly or
+    # use either function like `data_from_text`, `data_from_json` to populate data"""
+    data: Optional[str] = None
+    # Optional: Metadata associated with this record; can contain partition keys
+    # See - https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html
+    metadata: Optional[KinesisFirehoseResponseRecordMetadata] = None
+    _json_data: Optional[Any] = None
+    json_serializer: Callable = json.dumps
+    json_deserializer: Callable = json.loads
+
+    def data_from_byte(self, data: bytes):
+        """Populate data field using a byte like data"""
+        self.data = base64.b64encode(data).decode("utf-8")
+
+    def data_from_text(self, data: str):
+        """Populate data field using a string like data"""
+        self.data_from_byte(data.encode("utf-8"))
+
+    def data_from_json(self, data: Any):
+        """Populate data field using any structure that could be converted to json"""
+        self.data_from_text(data=self.json_serializer(data))
+
+    def asdict(self) -> Dict:
+        r: Dict[str, Any] = {
+            "recordId": self.record_id,
+            "result": self.result,
+            "data": self.data,
+        }
+        if self.metadata:
+            r["metadata"] = self.metadata.asdict()
+        return r
+
+    @property
+    def data_as_bytes(self) -> Optional[bytes]:
+        """Decoded base64-encoded data as bytes"""
+        if not self.data:
+            return None
+        return base64.b64decode(self.data)
+
+    @property
+    def data_as_text(self) -> Optional[str]:
+        """Decoded base64-encoded data as text"""
+        if not self.data_as_bytes:
+            return None
+        return self.data_as_bytes.decode("utf-8")
+
+    @property
+    def data_as_json(self) -> Optional[Dict]:
+        """Decoded base64-encoded data loaded to json"""
+        if not self.data_as_text:
+            return None
+        if self._json_data is None:
+            self._json_data = self.json_deserializer(self.data_as_text)
+        return self._json_data
+
+
+@dataclass
+class KinesisFirehoseResponse:
+    """Kinesis Data Firehose response object
+
+    Documentation:
+    --------------
+    - https://docs.aws.amazon.com/firehose/latest/dev/data-transformation.html
+
+    Parameters
+    ----------
+    records : List[KinesisFirehoseResponseRecord]
+        records of Kinesis Data Firehose response object,
+        optional parameter at start. can be added later using `add_record` function.
+    """
+
+    records: Optional[List[KinesisFirehoseResponseRecord]] = None
```

Comment:
```suggestion
    records: List[KinesisFirehoseResponseRecord] = field(default_factory=list)
```

You can create one by default, so you don't need yet another check and doesn't sacrifice readability ;)